src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CoreClrSocketConnection.cs

```diff
@@ -550,6 +550,20 @@ protected override void WriteCore(byte[] buffer, int offset, int size, bool imme
             }
         }
 
+        protected override void TraceWriteStart(int size, bool async)
+        {
+            var remoteEndpoint = (IPEndPoint)_socket.RemoteEndPoint;
+            string remoteEndpointAddressString = remoteEndpoint.Address + ":" + remoteEndpoint.Port;
```

Comment:
Is _socket guaranteed to be not null? I see TraceSocketReadStop below is checking for null.  Maybe a Contract.Assert here?